regression/cbmc/address_of_struct_member/address_of.c

```diff
@@ -0,0 +1,19 @@
+struct expr
+{
+  int line;
+  char comment;
+};
+
+int main(int argc, char *argv[])
+{
+  struct expr var_expr = {42, 'a'};
+  struct expr other_expr = {34, 'b'};
+  unsigned int nondet;
+  int *ref;
+  if(nondet)
+    ref = &var_expr.line;
+  else
+    ref = &other_expr.line;
+  __CPROVER_assert(*ref != 34, "expected failure: ref == 34");
```

Comment:
:pick: I am not sure this is a great test of the new SMT back end. The `line` field of `expr` will be at offset zero. It would be good to include taking the address of a field with a non-zero offset such as the `comment` field and to check that the resulting size of the offset is as expected.